test/test_autograd.py

```diff
@@ -2306,6 +2306,16 @@ def test_detach_base(self):
         self.assertIsNotNone(view.grad_fn)
         self.assertIs(view._base, x)
 
+    def test_detach_then_inplace_raises_in_autograd(self):
```

Comment:
What is this testing?
If you try to test the metadata change, you want to do `z.resize_()` or `z.as_strided_()` (any inplace view) and these should fail when called.

Reply:
I'll add an additional test for the metadata change, thanks for that suggestion

I think this test is testing that the version counter gets propagated? In a previous version of this PR, this test did not succeed

Reply:
The version counter should already be properly propagated in master. Maybe there is already a test for that? But if not, we definitely want one.

Reply:
There isn't a test for it (for detach). The version counter is indeed properly propagated in master